frontend/src/components/SemesterBlock.tsx

```diff
@@ -58,12 +63,24 @@ export class SemesterBlock extends React.Component<
   }
 
   renderBody() {
-    const status = this.props.semester.status;
+    const { semester, courseWarnings } = this.props;
+    const status = semester.status;
     if (status === "CLASSES" || status === "HOVERINACTIVE") {
-      return this.props.semester.classes.map((scheduleCourse, index) => {
+      return semester.classes.map((scheduleCourse, index) => {
         if (!!scheduleCourse) {
           return (
-            <ClassBlock key={index} class={scheduleCourse} index={index} />
+            <ClassBlock
+              key={index}
+              class={scheduleCourse}
+              index={index}
+              warning={
+                !!courseWarnings.find(
+                  w =>
+                    w.subject + w.classId ===
+                    scheduleCourse.subject + scheduleCourse.classId
```

Comment:
might also need to change this bit to either return the message, or store the entire warning